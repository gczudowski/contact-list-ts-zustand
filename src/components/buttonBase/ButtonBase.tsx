import React, { ReactElement } from 'react';

interface Props {
  onClick: () => void;
  label: string;
}

function ButtonBase({ onClick, label }: Props): ReactElement {
  return (
    <button style={styles.buttonBase} onClick={onClick} type="button">
      {label}
    </button>
  );
}

const styles = {
  buttonBase: {
    display: 'flex',
    width: '100%',
    minHeight: 36,
    fontSize: 18,
    backgroundColor: '#e74c3c',
    color: 'white',
    border: 'none',
    alignItems: 'center',
    justifyContent: 'center',
    cursor: 'pointer',
    boxShadow: '0px 1px 2px 0px rgb(0 0 0 / 15%)',
  },
};

export default ButtonBase;
