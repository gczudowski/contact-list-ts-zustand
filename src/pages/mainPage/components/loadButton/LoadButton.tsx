import React, { ReactElement } from 'react';
import { ScaleLoader } from 'react-spinners';
import ButtonBase from '@src/components/buttonBase/ButtonBase';

interface Props {
  isLoading: boolean;
  onClick: () => void;
}

function LoadButton({ isLoading, onClick }: Props): ReactElement {
  return (
    <div style={{ width: '100%' }}>
      {isLoading ? <ScaleLoader /> : <ButtonBase onClick={onClick} label="Load more" />}
    </div>
  );
}

export default LoadButton;
