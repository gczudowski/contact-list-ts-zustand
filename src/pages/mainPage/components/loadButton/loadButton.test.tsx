import React from 'react';
import renderer from 'react-test-renderer';
import ButtonBase from '@src/components/buttonBase/ButtonBase';
import LoadButton from './LoadButton';

jest.mock('react-spinners', () => ({
  __esModule: true,
  ScaleLoader: () => <div>loaderMock</div>,
}));

jest.mock('@src/components/buttonBase/ButtonBase', () => ({
  __esModule: true,
  default: () => <button type="button">buttonBaseMock</button>,
}));

describe('LoadButton', () => {
  const onClickCallback = () => {};

  describe('snapshots', () => {
    it('renders properly when isLoading is true', () => {
      const component = renderer.create(<LoadButton onClick={onClickCallback} isLoading />);

      expect(component.toJSON()).toMatchSnapshot();
    });

    it('renders properly when isLoading is false', () => {
      const component = renderer.create(<LoadButton onClick={onClickCallback} isLoading={false} />);

      expect(component.toJSON()).toMatchSnapshot();
    });
  });

  it('passes proper props to ButtonBase when isLoading is false', () => {
    const { root } = renderer.create(<LoadButton onClick={onClickCallback} isLoading={false} />);

    const button = root.findByType(ButtonBase);

    expect(button.props.onClick).toEqual(onClickCallback);
    expect(button.props.label).toEqual('Load more');
  });
});
