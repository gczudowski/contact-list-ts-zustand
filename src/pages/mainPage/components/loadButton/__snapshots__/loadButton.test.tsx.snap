// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadButton snapshots renders properly when isLoading is false 1`] = `
<div
  style={
    Object {
      "width": "100%",
    }
  }
>
  <button
    type="button"
  >
    buttonBaseMock
  </button>
</div>
`;

exports[`LoadButton snapshots renders properly when isLoading is true 1`] = `
<div
  style={
    Object {
      "width": "100%",
    }
  }
>
  <div>
    loaderMock
  </div>
</div>
`;
